test/cppsuite/test_harness/workload_validation.h

```diff
@@ -0,0 +1,395 @@
+        created_collections = parse_collections_schema(session, collection_name);
+
+        /* Allocate memory to the operations performed on the created collections. */
+        for (auto const &it : created_collections) {
+            std::map<int, std::string *> *map = new std::map<int, std::string *>();
+            collections[it] = map;
+        }
+
+        /*
+         * Build in memory the final state of each created collection according to the tracked
+         * operations.
+         */
+        collection_name = operation_table_name;
+        for (auto const &active_collection : created_collections)
+            parse_collections_operations(session, collection_name, active_collection, collections);
+
+        /* Check all tracked operations in memory against the database on disk. */
+        is_valid = check_reference(session, collections);
+
+        /* Check what has been saved on disk against what has been tracked. */
+        if (is_valid) {
+            for (auto const &collection : created_collections) {
+                is_valid = check_disk_state(session, collection, collections);
+                if (!is_valid) {
+                    debug_info("check_disk_state failed for collection " + collection, _trace_level,
+                      DEBUG_INFO);
+                    break;
+                }
+            }
+
+        } else
+            debug_info("check_reference failed!", _trace_level, DEBUG_INFO);
+
+        /* Clean the allocated memory. */
+        clean_memory(collections);
+
+        return is_valid;
+    }
+
+    /* Check what is present on disk against what has been tracked. */
+    bool
+    check_disk_state(WT_SESSION *session, const std::string &collection_name,
+      std::map<std::string, std::map<int, std::string *> *> &collections)
+    {
+        WT_CURSOR *cursor;
+        int key;
+        const char *value;
+        bool is_valid;
+        std::string *value_str;
+        std::map<int, std::string *> *collection;
+
+        testutil_check(session->open_cursor(session, collection_name.c_str(), NULL, NULL, &cursor));
+
+        /* Check the collection has been tracked and contains data. */
+        is_valid =
+          ((collections.count(collection_name) > 0) && (collections[collection_name] != nullptr));
+
+        if (!is_valid)
+            debug_info(
+              "Collection " + collection_name + " has not been tracked or has been deleted",
+              _trace_level, DEBUG_INFO);
+        else
+            collection = collections[collection_name];
+
+        /* Read the collection on disk. */
+        while (is_valid && (cursor->next(cursor) == 0)) {
+            testutil_check(cursor->get_key(cursor, &key));
+            testutil_check(cursor->get_value(cursor, &value));
+
+            debug_info("Key is " + std::to_string(key), _trace_level, DEBUG_INFO);
+            debug_info("Value is " + std::string(value), _trace_level, DEBUG_INFO);
+
+            value_str = (*collection)[key];
```

Comment:
Can we do `collection->at(key)`? I feel like that's cleaner than `(*collection)[key]`.